sdk/cosmos/azure-cosmos/azure/cosmos/aio/_container.py

```diff
@@ -88,33 +102,41 @@ async def is_system_key(self) -> bool:
             self._is_system_key = (
                 properties["partitionKey"]["systemKey"] if "systemKey" in properties["partitionKey"] else False
             )
-        return cast('bool', self._is_system_key)
+        return self._is_system_key
 
     @property
     def scripts(self) -> ScriptsProxy:
         if self._scripts is None:
             self._scripts = ScriptsProxy(self, self.client_connection, self.container_link)
-        return cast('ScriptsProxy', self._scripts)
+        return self._scripts
 
-    def _get_document_link(self, item_or_link: Union[Dict[str, Any], str]) -> str:
+    def _get_document_link(self, item_or_link: Union[str, Mapping[str, Any]]) -> str:
         if isinstance(item_or_link, str):
             return "{}/docs/{}".format(self.container_link, item_or_link)
         return item_or_link["_self"]
 
-    def _get_conflict_link(self, conflict_or_link: Union[Dict[str, Any], str]) -> str:
+    def _get_conflict_link(self, conflict_or_link: Union[str, Mapping[str, Any]]) -> str:
         if isinstance(conflict_or_link, str):
             return "{}/conflicts/{}".format(self.container_link, conflict_or_link)
         return conflict_or_link["_self"]
 
-    def _set_partition_key(self, partition_key) -> Union[str, Awaitable]:
+    async def _set_partition_key(
```

Comment:
I had to make this function async - which actually seems to have been the original intention, but needs confirmation.

Reply:
yep, this is the whole logic with awaiting the result of the container properties read result so it should've been async 👍